legend-depot-core-services/src/main/java/org/finos/legend/depot/services/projects/DependenciesCache.java

```
@@ -65,22 +66,30 @@ private void initCache()
         }
     }
 
-    private Function3<String,String,String,StoreProjectVersionData> getProjectDataFromStore()
+    private Function3<String,String,String,StoreProjectVersionData> getProjectVersionDataFromStore()
     {
-        return (group, artifact, versionId) -> this.projectsVersionsStore.find(group, artifact,versionId).get();
+        return (group, artifact, versionId) ->
+        {
+            Optional<StoreProjectVersionData> projectVersion = this.projectsVersionsStore.find(group, artifact, versionId);
+            if (!projectVersion.isPresent())
+            {
+                throw new RuntimeException(String.format("%s-%s-%s not found in store", group, artifact, versionId));
```

can we return an IllegalState exception.